csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -75,6 +76,22 @@ public static SessionOptions MakeSessionOptionWithCudaProvider(int deviceId = 0)
             return options;
         }
 
+        /// <summary>
+        /// A helper method to construct a SessionOptions object for TensorRT execution.
+        /// Use only if CUDA/TensorRT are installed and you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="deviceId"></param>
+        /// <returns>A SessionsOptions() object configured for execution on deviceId</returns>
+        public static SessionOptions MakeSessionOptionWithTensorrtProvider(int deviceId = 0)
+        {
+            CheckTensorrtExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_Tensorrt(options.Handle, deviceId));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_CUDA(options.Handle, deviceId));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_CPU(options.Handle, 1));
```

Comment:
We are not guarding against the options leak on exception here.

Reply:
looks like all the other MakeSessionOption static methods in this file are following the same pattern.
what's your suggestion. how should we recover here. this method returns options.
prevent the leak but propagate the exception?

Reply:
We just try/catch. In catch we do options.Dispose() and then rethrow.

Reply:
Need to fix the other location

Reply:
@yuslepukhin are you suggesting using try/catch as the previous discussion?  https://github.com/microsoft/onnxruntime/pull/7315#discussion_r613934184

Reply:
I've added try/catch around native methods.

Reply:
If you think others should also properly address this options leak, I can help modify in another PR.